synapse/tools/dmon.py

```diff
@@ -143,9 +143,11 @@ def main(argv, outp=None):
     p = getArgParser()
     opts = p.parse_args(argv)
 
-    if opts.log_level:
-        logging.basicConfig(level=opts.log_level.upper())
-        logger.info('log level set to ' + opts.log_level)
+    log_level = os.getenv('SYN_DMON_LOG_LEVEL', opts.log_level)
```

Comment:
Do we want to blow up if `SYN_DMON_LOG_LEVEL` is not in `LOG_LEVEL_CHOICES` ?

Reply:
I think so, if they go out of their way to specify a level, it should either just work, or break early so they can see the problem instead of having to figure out why stuff isn't being logged.

Reply:
this has changed to validate the envar value.